From e4a9d850bff80c6a4387f64468eb337335fae85a Mon Sep 17 00:00:00 2001
From: "Kevin F. Haggerty" <haggertk@lineageos.org>
Date: Fri, 30 Nov 2018 13:20:54 -0700
Subject: [PATCH 5/8] Allow dnsmasq to getattr netd unix_stream_socket

* Noted upon starting wifi tether
* avc: denied { getattr } for path="socket:[11154]" dev="sockfs"
  ino=11154 scontext=u:r:dnsmasq:s0 tcontext=u:r:netd:s0
  tclass=unix_stream_socket permissive=0

Change-Id: I98afc740fda3eb87a627abc414525b05f6877e33
---
 public/dnsmasq.te | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/public/dnsmasq.te b/public/dnsmasq.te
index 3aaefd3e..771c47e6 100644
--- a/public/dnsmasq.te
+++ b/public/dnsmasq.te
@@ -20,6 +20,6 @@ allow dnsmasq netd:fifo_file { read write };
 allow dnsmasq netd:netlink_kobject_uevent_socket { read write };
 allow dnsmasq netd:netlink_nflog_socket { read write };
 allow dnsmasq netd:netlink_route_socket { read write };
-allow dnsmasq netd:unix_stream_socket { read write };
+allow dnsmasq netd:unix_stream_socket { getattr read write };
 allow dnsmasq netd:unix_dgram_socket { read write };
 allow dnsmasq netd:udp_socket { read write };
-- 
2.17.1

